validphys2/src/validphys/closuretest/closure_results.py

```diff
@@ -394,3 +398,130 @@ def fit_underlying_pdfs_summary(fit, fitunderlyinglaw):
+                        for cd in commondata_wc]
+    modified_covmat = dataset_inputs_covmat_from_systematics(
+        commondata_wc_temp,
+        dataset_input_list,
+        use_weights_in_covmat=False,
+        norm_threshold=None,
+        _list_of_central_values=None,
+        _only_additive=False,
+    )
+
+    cov_dict["A/U"] = (or_trace-np.trace(modified_covmat))/or_trace*100
+    commondata_wc_temp = [cd.process_commondata(False,True,True,False,inconsistent_datasets,sys_rescaling_factor)
+                        for cd in commondata_wc]
+    modified_covmat = dataset_inputs_covmat_from_systematics(
+        commondata_wc_temp,
+        dataset_input_list,
+        use_weights_in_covmat=False,
+        norm_threshold=None,
+        _list_of_central_values=None,
+        _only_additive=False,
+    )
+    cov_dict["M/C"] = (or_trace-np.trace(modified_covmat))/or_trace*100
+
+    commondata_wc_temp = [cd.process_commondata(False,True,False,True,inconsistent_datasets,sys_rescaling_factor)
+                        for cd in commondata_wc]
+    modified_covmat = dataset_inputs_covmat_from_systematics(
+        commondata_wc_temp,
+        dataset_input_list,
+        use_weights_in_covmat=False,
+        norm_threshold=None,
+        _list_of_central_values=None,
+        _only_additive=False,
+    )
+    cov_dict["M/U"] = (or_trace-np.trace(modified_covmat))/or_trace*100
+
+    commondata_wc_temp = [cd.process_commondata(True,True,True,False,inconsistent_datasets,sys_rescaling_factor)
+                        for cd in commondata_wc]
+    modified_covmat = dataset_inputs_covmat_from_systematics(
+        commondata_wc_temp, 
+        dataset_input_list,
+        use_weights_in_covmat=False,
+        norm_threshold=None,
+        _list_of_central_values=None,
+        _only_additive=False,
+    )
+    cov_dict["AM/C"] = (or_trace-np.trace(modified_covmat))/or_trace*100
+
+    commondata_wc_temp = [cd.process_commondata(True,True,False,True,inconsistent_datasets,sys_rescaling_factor)
+                        for cd in commondata_wc]
+    modified_covmat = dataset_inputs_covmat_from_systematics(
+        commondata_wc_temp, 
+        dataset_input_list,
+        use_weights_in_covmat=False,
+        norm_threshold=None,
+        _list_of_central_values=None,
+        _only_additive=False,
+    )
+    cov_dict["AM/U"] = (or_trace-np.trace(modified_covmat))/or_trace*100
+
+    commondata_wc_temp = [cd.process_commondata(True,True,True,True,inconsistent_datasets,sys_rescaling_factor)
+                        for cd in commondata_wc]
+    modified_covmat = dataset_inputs_covmat_from_systematics(
+        commondata_wc_temp,
+        dataset_input_list,
+        use_weights_in_covmat=False,
+        norm_threshold=None,
+        _list_of_central_values=None,
+        _only_additive=False,
+    )
+    cov_dict["TOT"] = (or_trace-np.trace(modified_covmat))/or_trace*100
+
+    print(cov_dict)
```

Comment:
Of course remove here the print (I guess it was left here by mistake). Just a couple of general comments about this function:

1.  It seems you me that it is doing the same task again and again changing every time just a few parameters. If that is the case, create a function that does what you want and just call the function with different arguments. 
2. In general I did not fully understand what this functions is doing and how it is doing it. Please add some comments here and there to explain and give more telling names to the variables.